aws/resource_aws_budgets_budget.go

```diff
@@ -226,9 +331,77 @@ func resourceAwsBudgetsBudgetRead(d *schema.ResourceData, meta interface{}) erro
 
 	d.Set("time_unit", budget.TimeUnit)
 
+	return resourceAwsBudgetsBudgetNotificationRead(d, meta)
+}
+
+func resourceAwsBudgetsBudgetNotificationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+
+	accountID, budgetName, err := decodeBudgetsBudgetID(d.Id())
+
+	describeNotificationsForBudgetOutput, err := client.DescribeNotificationsForBudget(&budgets.DescribeNotificationsForBudgetInput{
+		BudgetName: aws.String(budgetName),
+		AccountId:  aws.String(accountID),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	notifications := make([]map[string]interface{}, 0)
+
+	for _, notificationOutput := range describeNotificationsForBudgetOutput.Notifications {
+		setNotificationDefaults(notificationOutput)
```

Comment:
We should remove this indirection as the function only handles one default, e.g.

```go
if notificationOutput.ThresholdType == nil {
	notification["threshold_type"] = budgets.ThresholdTypePercentage
} else {
	notification["threshold_type"] = aws.StringValue(notificationOutput.ThresholdType)
}
```